sdk/cosmos/azure_data_cosmos/src/handler/retry_handler.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use crate::retry_policies::BaseRetryPolicy;
+use async_trait::async_trait;
+use azure_core::http::{request::Request, RawResponse};
+use std::time::Duration;
+use tokio::time::sleep;
+
+// Helper trait to conditionally require Send on non-WASM targets
+#[cfg(not(target_arch = "wasm32"))]
+pub trait CosmosConditionalSend: Send {}
+#[cfg(not(target_arch = "wasm32"))]
+impl<T: Send> CosmosConditionalSend for T {}
+
+#[cfg(target_arch = "wasm32")]
+pub trait CosmosConditionalSend {}
+#[cfg(target_arch = "wasm32")]
+impl<T> CosmosConditionalSend for T {}
+
+/// Trait defining the interface for retry handlers in Cosmos DB operations
+///
+/// This trait provides a contract for implementing retry logic that wraps HTTP requests
+/// with automatic retry capabilities. Implementations can inject custom retry policies
+/// and handle both transient failures (errors) and non-success HTTP responses.
+#[allow(dead_code)]
+#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+pub trait AbstractRetryHandler: Send + Sync {
```

Comment:
Just `RetryHandler` would be the style here. Rust code generally doesn't refer to OOP terms like `Abstract`.